src/Dotnet.Script/Program.cs

```diff
@@ -232,20 +238,104 @@ private static int Wain(string[] args)
 
                 if (!string.IsNullOrWhiteSpace(file.Value))
                 {
-                    var optimizationLevel = OptimizationLevel.Debug;
-                    if (configuration.HasValue() && configuration.Value().ToLower() == "release")
+                    if (Debugger.IsAttached || nocache.HasValue())
                     {
-                        optimizationLevel = OptimizationLevel.Release;
+                        var optimizationLevel = OptimizationLevel.Debug;
+                        if (configuration.HasValue() && configuration.Value().ToLower() == "release")
+                        {
+                            optimizationLevel = OptimizationLevel.Release;
+                        }
+                        exitCode = await RunScript(file.Value, debugMode.HasValue(), logFactory, optimizationLevel, app.RemainingArguments.Concat(argsAfterDoubleHypen), interactive.HasValue(), packageSources.Values?.ToArray());
+                    }
+                    else
+                    {
+                        string cacheFolder = Path.Combine(Path.GetTempPath(), "dotnet-scripts");
```

Comment:
At some point we need to move all this logic out of `Program.cs` so that it can be used for people using `dotnet-script` programatically. This applies to everything here so we can do that later 👍

Reply:
I agree - let's keep this PR limited in size and put that in the next steps